ale/drivers/selene_drivers.py

```diff
@@ -435,3 +414,318 @@ def sensor_model_version(self):
+        """
+        return spice.gdpool('INS{}_DISTORTION_COEF_X'.format(self.ikid),0, 4).tolist()
+
+
+    @property
+    def _odky(self):
+        """
+        Returns the y coefficients of the optical distortion model.
+        Expects tc_id to be defined. This should be a string of the form
+        LISM_TC1 or LISM_TC2.
+
+        Returns
+        -------
+        : list
+          Optical distortion y coefficients
+        """
+        return spice.gdpool('INS{}_DISTORTION_COEF_Y'.format(self.ikid), 0, 4).tolist()
+
+    @property
+    def line_exposure_duration(self):
+        """
+        Returns Line Exposure Duration
+
+        Kaguya has an unintuitive key for this called CORRECTED_SAMPLING_INTERVAL.
+        The original LINE_EXPOSURE_DURATION PDS3 keys is often incorrect and cannot
+        be trusted.
+
+        Returns
+        -------
+        : float
+          Line exposure duration
+        """
+        # It's a list, but only sometimes.
+        # seems to depend on whether you are using the original zipped archives or
+        # if its downloaded from Jaxa's image search:
+        # (https://darts.isas.jaxa.jp/planet/pdap/selene/product_search.html#)
+        try:
+            return self.label['CORRECTED_SAMPLING_INTERVAL'][0].value * 0.001 # Scale to seconds
+        except:
+            return self.label['CORRECTED_SAMPLING_INTERVAL'].value * 0.001  # Scale to seconds
+
+
+    @property
+    def focal_length(self):
+        """
+        Returns camera focal length
+        Expects ikid to be defined. This should be the NAIF ID for the base band.
+
+        Returns
+        -------
+        : float
+          Camera focal length
+        """
+        return float(spice.gdpool('INS{}_FOCAL_LENGTH'.format(self.ikid), 0, 1)[0])
+
+    @property
+    def usgscsm_distortion_model(self):
+        """
+        Kaguya uses a unique radial distortion model so we need to overwrite the
+        method packing the distortion model into the ISD.
+
+        from the IK:
+
+        Line-of-sight vector of pixel no. n can be expressed as below.
+
+        Distortion coefficients information:
+        INS<INSTID>_DISTORTION_COEF_X  = ( a0, a1, a2, a3)
+        INS<INSTID>_DISTORTION_COEF_Y  = ( b0, b1, b2, b3),
+
+        Distance r from the center:
+        r = - (n - INS<INSTID>_CENTER) * INS<INSTID>_PIXEL_SIZE.
+
+        Line-of-sight vector v is calculated as
+        v[X] = INS<INSTID>BORESIGHT[X] + a0 + a1*r + a2*r^2 + a3*r^3 ,
+        v[Y] = INS<INSTID>BORESIGHT[Y] + r+a0 + a1*r +a2*r^2 + a3*r^3 ,
+        v[Z] = INS<INSTID>BORESIGHT[Z]
+
+        Expects odkx and odky to be defined. These should be a list of optical
+        distortion x and y coefficients respectively.
+
+        Returns
+        -------
+        : dict
+          radial distortion model
+
+        """
+        return {
+            "kaguyatc": {
+                "x" : self._odkx,
+                "y" : self._odky
+            }
+        }
```

Comment:
Needs to be tested, but based on Hiro's gist, TC and MI appear to use the same equations for the distortion model. Maybe we should rename this to the Kaguya distortion model instead of Kaguya TC